src/java.base/share/classes/jdk/internal/foreign/SegmentBulkOperations.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jdk.internal.foreign;
+
+import jdk.internal.misc.ScopedMemoryAccess;
+import jdk.internal.vm.annotation.ForceInline;
+
+/**
+ * This class contains optimized bulk operation methods that operate on one or several
+ * memory segments.
+ * <p>
+ * Generally, the methods attempt to work with as-large-as-possible units of memory at
+ * a time.
+ * <p>
+ * It should be noted that when invoking scoped memory access get/set operations, it
+ * is imperative from a performance perspective to convey the sharp types from the
+ * call site in order for the compiler to pick the correct Unsafe access variant.
+ */
+public final class SegmentBulkOperations {
+
+    private SegmentBulkOperations() {}
+
+    private static final ScopedMemoryAccess SCOPED_MEMORY_ACCESS = ScopedMemoryAccess.getScopedMemoryAccess();
+
+    // MISMATCH_NATIVE_THRESHOLD must be a power of two and should be greater than 2^3
+    private static final long MISMATCH_NATIVE_THRESHOLD = 1 << 20;
```

Comment:
Should we allow toggling this threshold via properties to allow using vectorizedMismatchLargeForBytes for small lengths, such as for testing or debugging?